tools/2024-09-gitops-flux.md

```diff
@@ -0,0 +1,289 @@
+
+### Architecture Diagram
+<!--
+Provide a diagram of the system architecture, illustrating how different
+components interact with each other in the context of this proposal.
+
+Include separate high level architecture diagram and component specific diagrams, wherever appropriate.
+-->
+
+![Architecture Diagram](2024-09-gitops-flux/architecture.png)
+
+### Detailed Design
+
+<!--
+This section should be detailed and thorough enough that another developer
+could implement your design and provide enough detail to get a high confidence
+estimate of the cost to implement the feature but isn’t as detailed as the 
+code. Be sure to also consider testability in your design.
+
+For each change, give each "change" in the proposal its own section and
+describe it in enough detail that someone else could implement it. Cover
+ALL of the important decisions like names. Your goal is to get an agreement
+to proceed with coding and PRs.
+
+If there are alternatives you are considering please include that in the open
+questions section. If the product has a layered architecture, it's good to
+align these sections with the product's layers. This will help readers use
+their current understanding to understand your ideas.
+
+Discuss the rationale behind architectural choices and alternative options 
+considered during the design process.
+-->
+
+#### Flux Controller
+
+> Note: this feature area hasn't been fully designed yet. This is a placeholder for the detailed design. Let's brainstorm in the design meeting if we have time.
+
+The Radius Flux Controller will be a Kubernetes controller, based on the [Flux source watcher](https://fluxcd.io/flux/gitops-toolkit/source-watcher/) pattern to watch for changes to `.bicep` and `.bicepparam` files in the Flux source controller. When a change is detected, the controller will compile the Bicep manifests and create or update `BicepDeployment` resources on the Kubernetes cluster. The controller is responsible for compiling Bicep manifests, so it will need to construct a "file system" that the Bicep compiler can use to resolve dependencies. We will need to think about how to handle these resources in an async manner, probably using a pattern and control loop similar to the [Flux Kustomize Controller](https://fluxcd.io/flux/components/kustomize/) design.
```

Comment:
> so it will need to construct a "file system" that the Bicep compiler can use to resolve dependencies

will it also have the ability to detect when `.bicep` or `.bicepparam` files are removed to trigger deletions or modifications?